src/im_context_simple.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use IMContextSimple;
+
+use IsA;
+use ffi;
+use glib::translate::*;
+
+use std::path::Path;
+
+pub trait IMContextSimpleExtManual {
+    fn add_compose_file<P: AsRef<Path>>(&self, compose_file: P);
+    fn add_table(&self, data: &[u16], max_seq_len: u32, n_seqs: u32);
+}
+
+impl<O: IsA<IMContextSimple>> IMContextSimpleExtManual for O {
+    fn add_compose_file<P: AsRef<Path>>(&self, compose_file: P) {
+        unsafe {
+            let compose_file = compose_file.as_ref();
+            ffi::gtk_im_context_simple_add_compose_file(self.to_glib_none().0, compose_file.to_glib_none().0);
+        }
+    }
+
+    fn add_table(&self, data: &[u16], max_seq_len: u32, n_seqs: u32) {
```

Comment:
@sdroege: So I did it like this. Should we instead compute `n_seqs` from the `data` length or just keep it like this?

Reply:
I'll check in detail on Monday, also your question about the closure/accelgroup

Reply:
I don't actually understand how that API works. This adds a whole table, so the data is a table. The docs say that `Each row of the table consists of max_seq_len key symbols followed by two guint16 interpreted as the high and low words of a gunicode value.`. Now how do we know how long each row in the data is?

Once we know how that API works, it might be better to instead pass a (repr packed!)  slice of `struct TableRow { symbols: XX, value: [u16; 2] }`.